app/controllers/tags_controller.rb

```diff
@@ -24,32 +24,49 @@ def load_tag_and_subtags
     end
   end
 
+  def reindex
```

Comment:
I've never had the pleasure of reindexing the works for a tag, but this looks reasonable! Is there someone else who could review it with a bit more authority?